src/lib/gradientParser.test.ts

```diff
@@ -0,0 +1,288 @@
+import { parseGradient } from "./gradientParser"
+
+test("parseGradient(linear-gradient)", () => {
+  expect(parseGradient("linear-gradient(180deg, red)")).toStrictEqual(
+    parseGradient("linear-gradient(rgb(255, 0, 0))")
+  )
+  expect(parseGradient("linear-gradient(blue 0)")).toStrictEqual(
+    parseGradient("linear-gradient(blue 0px)")
+  )
+  expect(parseGradient("linear-gradient(to left, red, blue)")).toMatchInlineSnapshot(`
+    [
+      {
+        "colorStops": [
+          {
+            "rgba": {
+              "a": 1,
+              "b": 0,
+              "g": 0,
+              "r": 255,
+            },
+            "type": "color-stop",
+          },
+          {
+            "rgba": {
+              "a": 1,
+              "b": 255,
+              "g": 0,
+              "r": 0,
+            },
+            "type": "color-stop",
+          },
+        ],
+        "gradientLine": {
+          "type": "side-or-corner",
+          "value": "left",
+        },
+        "type": "linear-gradient",
+      },
+    ]
+  `)
+  expect(parseGradient("linear-gradient(0.25turn, #fff 10px, 50%, red)")).toMatchInlineSnapshot(`
+    [
+      {
+        "colorStops": [
+          {
+            "position": {
+              "unit": "px",
+              "value": 10,
+            },
+            "rgba": {
+              "a": 1,
+              "b": 255,
+              "g": 255,
+              "r": 255,
+            },
+            "type": "color-stop",
+          },
+          {
+            "hint": {
+              "unit": "%",
+              "value": 50,
+            },
+            "type": "color-hint",
+          },
+          {
+            "rgba": {
+              "a": 1,
+              "b": 0,
+              "g": 0,
+              "r": 255,
+            },
+            "type": "color-stop",
+          },
+        ],
+        "gradientLine": {
+          "type": "angle",
+          "value": 90,
+        },
+        "type": "linear-gradient",
+      },
+    ]
+  `)
+})
+
+test("parseGradient(radial-gradient)", () => {
+  expect(parseGradient("radial-gradient(5em circle at top left, yellow, blue)"))
+    .toMatchInlineSnapshot(`
+      [
+        {
+          "colorStops": [
+            {
+              "rgba": {
+                "a": 1,
+                "b": 0,
+                "g": 255,
+                "r": 255,
+              },
+              "type": "color-stop",
+            },
+            {
+              "rgba": {
+                "a": 1,
+                "b": 255,
+                "g": 0,
+                "r": 0,
+              },
+              "type": "color-stop",
+            },
+          ],
+          "endingShape": "circle",
+          "position": "top left",
+          "size": [
+            {
+              "unit": "em",
+              "value": 5,
+            },
+          ],
+          "type": "radial-gradient",
+        },
+      ]
+    `)
+  expect(
+    [
+      "ellipse farthest-corner",
+      // 'ellipse cover',
```

Comment:
those are supposed to be supported later?

Reply:
found those in gradient-parser tests, but they aren't part of the spec afaict, nor are they supported in chrome, so I didn't fix them